polars/polars-core/src/chunked_array/temporal/datetime.rs

```diff
@@ -11,6 +11,19 @@ use super::*;
 use crate::prelude::DataType::Datetime;
 use crate::prelude::*;
 
+#[cfg(feature = "timezones")]
+fn validate_time_zone(tz: TimeZone) {
+    use arrow::temporal_conversions::parse_offset;
+    use chrono_tz::Tz;
+    match parse_offset(&tz) {
+        Ok(_) => (),
+        Err(_) => match tz.parse::<Tz>() {
+            Ok(_) => (),
+            Err(_) => panic!("time zone {tz} not supported"),
```

Comment:
We should return an error here and every function that calls this.

The return type will be `PolarsResult<()>` and end the function with `Ok(())`